client/src/pages/RegisterPage/RegisterPage.js

```diff
@@ -16,15 +18,42 @@ const Register = ({ setAlert, register, isAuthenticated, alert, guestMode, toggl
     password2: '',
   })
 
+  const dispatch = useDispatch()
+  const message = useSelector(state => state.auth.message)
+  const error = useSelector(state => state.auth.error)
+
   const { name, email, password, password2 } = formData
 
+  // listen for errors from our server and display them, then reset in
+  useEffect(()=>{
+    if (error) {
+      showToast(error, 'warn')
+      resetAuthMessages(dispatch)
+    }
+  }, [error])
+
+  // listen to messages from our server and display them, then reset it
+  useEffect(()=>{
+    if (message) {
+      showToast(message, 'success')
+      resetAuthMessages(dispatch)
+    }
+  }, [message])
```

Comment:
Hmm I feel like we should be able to do this without a useEffect and tie it instead to the actual register function calls